new-library.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+require 'fileutils'
+
+def cp(src, dest, app_name)
+  if File.directory?(src)
+    Dir.mkdir(dest) unless Dir.exists?(dest)
+  else
+    content = File.read(src)
+      .gsub("Sample", app_name)
+    File.write(dest, content)
+  end
+end
+
+def main(dest, app_name)
+  source = File.expand_path("../Libraries/Sample", __FILE__)
+  files = Dir.chdir(source) { Dir["**/*"] }
+    .reject { |file| file["project.xcworkspace"] || file["xcuserdata"] }
+    .each { |file|
+      new_file = file.gsub("Sample", app_name)
+      cp File.join(source, file), File.join(dest, new_file), app_name
+    }
+end
+
+if ARGV.count == 0
+  puts "Usage: #{__FILE__} <LibraryNameInCamelCase>"
+  puts ""
+  puts "This script will bootstrap a new React Native library for use in the current project"
+else
+  app_name = ARGV.first
+  dest = File.expand_path("Libraries/#{app_name}/")
+  abort "#{dest} already exists." if Dir.exists?(dest)
+
+  FileUtils.mkdir_p(dest)
+  puts "Setting up new React Native library in #{dest}"
+  puts ""
+
+  main(dest, app_name)
+
+  puts "Next steps:"
+  puts ""
+  puts "   Open #{app_name}.xcproject in Xcode"
+  puts "   Link this project to your main .xcproject"
```

Comment:
I'm adding it to docs on #300, just making some images as @frantic suggested, maybe it could link to the docs.